walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -28,11 +28,27 @@
       return new ArrayList<TimeRange>();
     }
     
-    List<TimeRange> unavailableTimes = getSortedUnavailableTimes(events, request);
-    return computeAvailableTimes(unavailableTimes, request);
+    // The Collection that stores the available times for the requested meeting.
+    Collection<TimeRange> availableTimes;
+
+    List<TimeRange> unavailableTimesWithOptionals = getSortedUnavailableTimes(events, request, true);
+    Collection<TimeRange> timesWithOptionals = computeAvailableTimes(unavailableTimesWithOptionals, request);
+
+    List<TimeRange> unavailableTimesWithoutOptionals = getSortedUnavailableTimes(events, request, false);
+    Collection<TimeRange> timesWithoutOptionals = computeAvailableTimes(unavailableTimesWithoutOptionals, request);
+
+    if (timesWithOptionals.isEmpty() && (request.getAttendees().size() == 0)) {
```

Comment:
How about this (we prefer these pattern to if-elses because it makes things more readable/simpler):
if (request.getAttendees().isEmpty()) {
  return timesWithOptionals;
}
if (!timesWithOptionals.isEmpty()) {
  return timesWithOptionals;
}
return timesWithoutOptionals;

Reply:
Ack. I won't use if else patterns from now on.

Reply:
1. Any reason you use size() == 0 instead of isEmpty for getAttendees? In general, you should prefer the operation that fits better for the purpose (which is isEmpty in this case).
2. Why do you need timesWithOptionals.isEmpty() check in the beginning? Yours and mine are identical, but my suggestion is simpler and more intuitive (IMHO).

Reply:
Oh I'm sorry, I thought I had made the change you suggested already. I must have missed that before pushing my last commit.